From bb4863f0ba25411ca546c586dcbb6f77f359597b Mon Sep 17 00:00:00 2001
From: Nicolai Buchwitz <nb@tipi-net.de>
Date: Thu, 28 May 2026 12:27:53 +0200
Subject: [PATCH] arm64/configs: Enable Microchip KSZ DSA support

Enable DSA support for Microchip KSZ switches with SPI interface. It is
for example used in the CM4S based RevPi Flat S.

Signed-off-by: Nicolai Buchwitz <nb@tipi-net.de>
---
 arch/arm64/configs/bcm2711_defconfig    | 2 ++
 arch/arm64/configs/bcm2711_rt_defconfig | 2 ++
 arch/arm64/configs/bcm2712_defconfig    | 2 ++
 3 files changed, 6 insertions(+)

--- a/arch/arm64/configs/bcm2711_defconfig
+++ b/arch/arm64/configs/bcm2711_defconfig
@@ -526,6 +526,8 @@ CONFIG_VETH=m
 CONFIG_NETKIT=y
 CONFIG_NET_VRF=m
 CONFIG_VSOCKMON=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_COMMON=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_SPI=m
 CONFIG_BCMGENET=y
 CONFIG_MACB=y
 CONFIG_IGB=m
--- a/arch/arm64/configs/bcm2711_rt_defconfig
+++ b/arch/arm64/configs/bcm2711_rt_defconfig
@@ -526,6 +526,8 @@ CONFIG_VETH=m
 CONFIG_NETKIT=y
 CONFIG_NET_VRF=m
 CONFIG_VSOCKMON=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_COMMON=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_SPI=m
 CONFIG_BCMGENET=y
 CONFIG_MACB=y
 CONFIG_IGB=m
--- a/arch/arm64/configs/bcm2712_defconfig
+++ b/arch/arm64/configs/bcm2712_defconfig
@@ -529,6 +529,8 @@ CONFIG_VETH=m
 CONFIG_NETKIT=y
 CONFIG_NET_VRF=m
 CONFIG_VSOCKMON=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_COMMON=m
+CONFIG_NET_DSA_MICROCHIP_KSZ_SPI=m
 CONFIG_BCMGENET=y
 CONFIG_MACB=y
 CONFIG_IGB=m
